src/lib/version.rs

```diff
@@ -0,0 +1,85 @@
+use serde::{Deserialize, Serialize};
+use serde_yaml;
+use std::fs;
+
+pub const VERSION_FILE: &str = "version.yaml";
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct VersionInfo {
+    pub commit: String,
+    pub source: String,
+    pub version: String,
+}
+
+pub fn read_version(handle: &str) -> VersionInfo {
```

Comment:
Another way to do this would be with `handle: Option<&str>` and then if None then use VERSION_FILE and then you don't have to import and pass it back everywhere but still have the testability. We can leave it like this for this PR, just an observation.